engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java

```diff
@@ -139,15 +141,17 @@ public synchronized final Changes evaluateScript(final String script, @Nullable
 
         RuntimeException evaluateErr = null;
         final Changes diff;
+
         // retain any objects which are created in the executed code, we'll release them when the script session
         // closes
         try (final S initialSnapshot = takeSnapshot();
-                final SafeCloseable ignored = LivenessScopeStack.open(this, false)) {
+                final SafeCloseable ignored = LivenessScopeStack.open(queryScope, false)) {
 
             try {
                 // Actually evaluate the script; use the enclosing auth context, since AbstractScriptSession's
                 // ExecutionContext never has a non-null AuthContext
                 executionContext.withAuthContext(ExecutionContext.getContext().getAuthContext())
+                        .withQueryScope(queryScope)
```

Comment:
It might not be necessary today, that's correct, but if not needed, it will be a no-op. Where possible (and straightforward, and obviously not-incorrect), Ryan and I were trying to make this ready for multiple script sessions, so if it does nothing, I'd be inclined to keep it.